standards/X509-SVID.md

```diff
@@ -87,12 +89,14 @@ This section describes how an X.509 SVID is validated. The procedure uses standa
 ### 5.1. Path Validation
 The validation of trust in a given SVID is based on standard X.509 path validation, and MUST follow [RFC 5280][1] path validation semantics.
 
-Certificate path validation requires the leaf SVID certificate and one or more SVID signing certificates. The set of signing certificates required for validation is known as the CA bundle. The mechanism through which an entity can retrieve the relevant CA bundle(s) is out of scope for this document, and is instead defined in the SPIFFE Workload API specification.
+Certificate path validation requires the leaf SVID certificate and one or more SVID signing certificates. The set of signing certificates required for validation is known as the CA bundle. While a leaf SVID's signing certificate must itself be an SVID, validators MAY skip SVID-specific checks on signing certificates due to the difficulties that would be introduced in leveraging existing path building logic/libraries.
```

Comment:
While I understand the concern that validation of SVIDs is currently not widely supported, it's not clear to me why skipping signing cert validation is safe. Instead of baking in an exception for validation here, should we focus efforts on broad validation support? I'd also be open to discussing the specific difficulties seen in implementing path validation.

Reply:
First some background (apologies, you may already be aware of this, but stating anyways for completeness): A previous validation mechanism stated that the immediate issuing CA of an SVID must reside in the same trust domain as the SVID it issues... that step was redacted when it was discovered that it could easily be broken by minting a new CA certificate with the desired trust domain... this was essentially a poor-persons attempt at supporting name constraints. To rectify the problem, it was decided that the validation should simply be "here are the roots for trust domain X". This paradigm is exhibited in the workload api, but unfortunately, is not well documented in X509-SVID (maybe we should add a blurb about this??)

After this change, retaining the SVID status of a signing certificate seems... somewhat cosmetic. There is at least _some_ value in the current signing certificate SVID spec, mainly considerations that help prevent it from being used as a workload identity. For that reason, I think it makes sense to keep a MUST for `A signing certificate is itself an SVID`. That being said, it doesn't really break anything if a signing cert is _not_ an SVID.

Now, validating that a given signing certificate _is_ an SVID could be difficult in practice. This is because there may be many valid certification paths, and not all libraries expose the full set of chains/paths. For instance, a leaf may exist for which there are two valid issuers (sharing a DN and public key)... one of the issuers may be a valid SVID, and the other may not. In order to perform SVID validation of the issuer, the client will need to have access to _all_ discovered chains, and not just one. Golang supports this, but I am worried that making this a MUST is a deal breaker for some.

I took a page out of 5280's book in allowing validators to skip this step... there are a few places there where issuers are constrained by a MUST, but validators aren't required to check it... It's usually done to ensure that issuers support a certain feature, while allowing validators to use the feature if they so choose. This is a little different of course, but the approach still made sense to me for this case.

Maybe a compromise here could be:
`validators MAY skip SVID-specific checks on signing certificates where it is not possible to do so`? Curious to hear your thoughts on this

Reply:
This line is no longer valid, it should be removed from this PR